More work on the story mode

alexpearce opened the pull request:
See issue #137 for motivation.

This PR begins the reworking of the first lessons, up to and including “Fun with LoKi Functors”. This means changes to the style and tone to make the lessons more approachable and more consistent with each other, and also adding challenges that should be given to the students during the lessons.

I'll push more changes to this branch for the other lessons.

alexpearce committed:
Add LaTeX support with KaTeX.

Just use LaTeX as normal in your Markdown with $ and $$.

alexpearce committed:
Add missing image file.

alexpearce committed:
Tidy up interactive DST Python script.

Closes issue #141.

alexpearce committed:
Rework bookkeeping lesson.

alexpearce committed:
Rework grid file download lesson.

alexpearce committed:
Rework LoKi functors lesson.

alexpearce committed:
Add challenge to data flow lesson.

alexpearce committed:
Add motivation to minimal DV job lesson.

alexpearce committed:
Tweak some titles.

alexpearce committed:
Remove old note from index.

alexpearce committed:
Reorder DaVinci lesson in index.

alexpearce committed:
Tweak course introduction.

alexpearce committed:
PEP8 compliance.

alexpearce committed:
Don’t use relative URLs.

It’s usually a good practice, but it won’t work when developing and viewing the files with the file:// protocol.

alexpearce commented:
One cool this PR also does is add KaTeX. Check out the “Finding data in the Bookkeeping” lesson for an example:

<img width="950" alt="LaTeX up in ur lesson" src="https://cloud.githubusercontent.com/assets/540853/11098479/90f0d93a-88a5-11e5-90ed-ff638d3385cb.png">

betatim commented:
Are we trying to merge this before tomorrow morning?

betatim commented on 03-bookkeeping.md:
```diff
@@ -35,38 +46,58 @@ representation of the [event
-will find a list of different configurations depending on the
-magnet polarity (`MagDown` and `MagUp`) as well as different
-versions of `Pythia` used to generate the events. We will use
+will find a several different subfolders to choose from. The names of these 
```
find a several -> find several

alexpearce replied:
Fixed.

betatim commented on 05-interactive-dst.md:
```diff
@@ -23,36 +23,19 @@ is not working as expected.
 and so on. Here we assumed the file you downloaded is called `00035742_00000002_1.allstreams.dst`.
-To take a look at the contents of the TES we need a simple
-options file:
+To take a look at the contents of the TES we to write a small
```
wut?

alexpearce replied:
Should have been “To take a look at the contents of the TES, we need to write a small…”. Fixed.

alexpearce commented:
Ideally yeah.

> On 11 Nov 2015, at 22:16, Tim Head notifications@github.com wrote:
> 
> Are we trying to merge this before tomorrow morning?
> 
> —
> Reply to this email directly or view it on GitHub.

alexpearce committed:
Fix typos.

apuignav commented:
Great work! I've checked the changes, apart from the comment on the centrally downloaded file I'm happy to merge. I would like another +1, though.

alexpearce committed:
Add instructions for downloading file from EOS.

Lots of people downloading from the Grid during the physical Starterkit lessons can be very slow. If we ‘cache’ the file on EOS we can use that
instead.

betatim commented:
I've not had time to read all of it ... if you want to use it for the lesson today coordinate with the other instructor and merge it if they are fine with the changes.

roelaaij commented on 04-files-from-grid.md:
```diff
@@ -44,7 +48,14 @@ dirac-dms-get-file LFN:/lhcb/MC/2012/ALLSTREAMS.DST/00035742/0000/00035742_00000
 
-Since those tuples tend to be quite large, you might want to use your AFS work directory instead of your AFS user directory to store files.
+> ## Downloading the file during a Starterkit lesson {.callout}
+> Lots of people downloading the same file at the same time can be very slow. As a workout, the file is also available on EOS, and can be downloaded to your current directory with the following command:
```
workout -> workaround
I do love them workouts :)

alexpearce replied:
Good spot! I've fixed it.

roelaaij commented:
I've had a look and I think this is great. I'm the other instructor this morning and I'd be very happy to use your changes. If there is still time, I'd propose to merge.

alexpearce committed:
Typo: workout -> workaround.

alexpearce commented:
Awesome, thanks everyone!
